code/src/OSFramework/Grid/AbstractDataSource.ts

```diff
@@ -227,11 +227,25 @@ namespace OSFramework.Grid {
         }
 
         public getRowNumberByKey(key: string): number {
-            return this.parentGrid.provider.rows.findIndex(
-                (item) =>
-                    _.get(item.dataItem, this.parentGrid.config.keyBinding) ===
-                    key
-            );
+            // Throws the error when is invalid
+            try {
+                const row = this.parentGrid.provider.rows.findIndex(
+                    (item) =>
+                        _.get(
+                            item.dataItem,
+                            this.parentGrid.config.keyBinding
+                        ).toString() === key
+                );
+
+                // Validation of row to prevent the default row key
+                if (row === -1) {
+                    throw new Error(Enum.ErrorMessages.Row_InvalidRowDataKey);
+                }
```

Comment:
This throw will be lost... since it will be caught right after... This doesn't make much sense.

Reply:
Fixed